providers/redfish/firmware.go

```diff
@@ -43,6 +48,14 @@ func (c *Conn) FirmwareInstall(ctx context.Context, component, applyAt string, f
 		return "", errors.Wrap(bmclibErrs.ErrFirmwareInstall, "invalid applyAt parameter: "+applyAt)
 	}
 
+	// expect atleast 10 minutes left in the deadline to proceed with the update
+	//
+	// this gives the BMC enough time to have the firmware uploaded and return a response to the client.
+	ctxDeadline, _ := ctx.Deadline()
+	if time.Until(ctxDeadline) < 10*time.Minute {
+		return "", errors.Wrap(errInsufficientCtxTimeout, " "+time.Until(ctxDeadline).String())
```

Comment:
What's the operational impact of this error? Is this something we expect to have retried? Would we expect a retry to work? Does an exit here leave the BMC in a sane state?

Reply:
As in with all bmclib functions, its upto the caller to retry with a higher timeout value and/or to close the bmc connection. We provide examples for invoking these methods - https://github.com/bmc-toolbox/bmclib/blob/main/examples/install-firmware/main.go